main.rb

```diff
@@ -0,0 +1,141 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    array = to_a
+    i = 0
+    while i < array.length
+      yield(array[i])
+      i += 1
+    end
+    array
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    array = to_a
+    i = 0
+    while i < array.length
+      yield(array[i], i)
+      i += 1
+    end
+    array
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each { |elem| new_array.push(elem) if yield(elem) == true }
+    new_array
+  end
+
+  def my_all?
+    return to_enum(:my_all?) unless block_given?
+
+    my_each { |elem| return false if yield(elem) == false }
+    true
+  end
+
+  def my_any?
+    return to_enum(:my_any?) unless block_given?
+
+    my_each { |elem| return true if yield(elem) == true }
+    false
+  end
+
+  def my_none?
+    return to_enum(:my_none?) unless block_given?
+
+    my_each { |elem| return false if yield(elem) == true }
+    true
+  end
+
+  def my_count(num = nil)
+    counter = 0
+    if num
+      my_each { |elem| counter += 1 if elem == num }
+
+    elsif block_given?
+      my_each { |elem| counter += 1 if yield(elem) }
+
+    else
+      counter = size
+    end
+    counter
+  end
+
+  def my_map(proc = nil)
+    return to_enum(:my_map) unless block_given?
+
+    new_array = []
+    if proc
+      my_each { |i| new_array.push(proc.call(i)) }
+    else
+      my_each { |i| new_array.push(yield(i)) }
+    end
+    new_array
+  end
+
+  def my_inject(num = nil, symbol = nil)
+    if block_given?
+      acc = num
+      my_each { |i| acc = acc.nil? ? i : yield(acc, i) }
+      acc
+    elsif acc.nil?
+      !num.nil? && num.is_a?(Symbol) || num.is_a?(String)
+      my_each { |i| acc = acc.nil? ? i : acc.send(num, i) }
+    elsif (acc = num)
+      !num.symbol? && num.is_a?(Symbol) || num.is_a?(String)
+      my_each { |i| acc = acc.nil? ? i : acc.send(symbol, i) }
+    end
+    acc
+  end
+
+  def multiply_els
+    my_inject { |x, y| x * y }
+  end
+end
```

Comment:
Enumerable `multiply_els` accepts an array as an argument and multiplies all the elements of the array together using #my_inject
      Failure/Error: `actual = multiply_els [2, 4, 5]`

      NoMethodError:
        undefined method `multiply_els'